tests/RobotFramework/tests/cumulocity/monitoring/monitor_device_collectd.robot

```diff
@@ -0,0 +1,52 @@
+*** Settings ***
+Documentation      With thin-edge.io device monitoring, you can collect metrics from your device 
+...                and forward these device metrics to IoT platforms in the cloud. Using these metrics, 
+...                you can monitor the health of devices and can proactively initiate actions in case the 
+...                device seems to malfunction. Additionally, the metrics can be used to help the customer 
+...                troubleshoot when problems with the device are reported. Thin-edge.io uses the open source 
+...                component collectd to collect the metrics from the device. Thin-edge.io translates the 
+...                collected metrics from their native format to the thin-edge.io JSON format and then into 
+...                the cloud-vendor specific format.
+...                Enabling monitoring on your device is a 3-steps process:
+...                1. Install collectd,
+...                2. Configure collectd,
+...                3. Enable thin-edge.io monitoring.
+
+
+Resource        ../../../resources/common.resource
+Library         ThinEdgeIO
+Library         Cumulocity
+
+Suite Setup     Custom Setup
+Test Teardown    Get Logs
+
+Force Tags      theme:monitoring    theme:c8y    theme:collectd
+
+*** Test Cases ***
+
+Check running collectd
+    Service Should Be Running    collectd
+    
+Is collectd publishing MQTT messages?
+    ${messages}=    Should Have MQTT Messages    topic=collectd/#    minimum=1    maximum=None
+    # Should Contain    ${messages[0]}    "pid":${pid}
+    # Should Contain    ${messages[0]}    "status":"up"
+
+Check thin-edge monitoring
+    # 3. Enable thin-edge.io monitoring.
+    Execute Command    sudo systemctl enable tedge-mapper-collectd
+    Execute Command    sudo systemctl start tedge-mapper-collectd
+    # Check thin-edge monitoring
+    ${tedge_messages}=    Should Have MQTT Messages    topic=tedge/measurements    minimum=1    maximum=None
+    ${c8y_messages}=    Should Have MQTT Messages    topic=c8y/measurement/measurements/create    minimum=1    maximum=None
```

Comment:
It's okay to only check some collectd messages are sent and translated, but it would be good to also check the grouping done by the collectd mapper. I fully understand that using the uncontrolled collectd output is not feasible, but we can send fake messages on collectd topics (as this was done by [this PySys test](https://github.com/thin-edge/thin-edge.io/blob/main/tests/PySys/monitoring/monitoring_with_simulated_messages/run.py))